synapse/storage/devices.py

```diff
@@ -284,6 +285,8 @@ def _get_devices_by_remote_txn(self, txn, destination, from_stream_id,
 
         results = []
         for user_id, user_devices in devices.iteritems():
+            # We bind literal True, as its database dependent how booleans are
+            # handled.
```

Comment:
I might prefer something more along the lines of.

```
# postgresql and sqlite use different formats for boolean literals, so we pass True as a query parameter rather than including a literal True in the query string itself
```